system/spd.c

```diff
@@ -0,0 +1,977 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (C) 2004-2023 Sam Demeulemeester
+
+#include "stdbool.h"
+#include "stdint.h"
+#include "string.h"
+
+#include "smbus.h"
+#include "spd.h"
+#include "jedec_id.h"
+#include "print.h"
+
+/** Rounding factors for timing computation
+ *
+ *  These factors are used as a configurable CEIL() function
+ *  to get the upper int from a float past a specific decimal point.
+ */
+
+#define DDR5_ROUNDING_FACTOR    30
+#define ROUNDING_FACTOR         0.9f
+
+ram_info_t ram = { 0, 0, 0, 0, 0, 0, "N/A"};
+
+static inline uint8_t bcd_to_ui8(uint8_t bcd)
+{
+    return bcd - 6 * (bcd >> 4);
+}
+
+void print_spdi(spd_info spdi, uint8_t row)
```

Comment:
Agree with Martin that `print_spdi` should be moved to app ultimatly, but that could be done later.

Reply:
OK, will handle this post LA64 merge.